lib/origen_testers/smartest_based_tester/base.rb

```diff
@@ -338,9 +338,18 @@ def pattern_header(options = {})
         }.merge(options)
         pin_list = ordered_pins.map do |p|
           if Origen.app.pin_pattern_order.include?(p.id)
-            p.id.to_s.upcase # specified name overrides pin name
+            # specified name overrides pin name
+            if p.is_a?(Origen::Pins::PinCollection) || p.id != p.name
```

Comment:
Just to the first part right?  (before the ||).

From: Stephen McGinty [mailto:notifications@github.com]
Sent: Thursday, April 21, 2016 9:04 AM
To: Origen-SDK/origen_testers <origen_testers@noreply.github.com>
Cc: chrisnappi <chris.nappi@freescale.com>; Author <author@noreply.github.com>
Subject: Re: [Origen-SDK/origen_testers] Allow v93k header to use lower case for groups and aliases (#11)


In lib/origen_testers/smartest_based_tester/base.rb<https://github.com/Origen-SDK/origen_testers/pull/11#discussion_r60585418>:

> @@ -338,9 +338,18 @@ def pattern_header(options = {})

>          }.merge(options)

>          pin_list = ordered_pins.map do |p|

>            if Origen.app.pin_pattern_order.include?(p.id)

> -            p.id.to_s.upcase # specified name overrides pin name

> +            # specified name overrides pin name

> +            if p.is_a?(Origen::Pins::PinCollection) || p.id != p.name

Can we add a size > 1 qualifier to the pin collection case?

Not sure it would ever be hit in reality, but conceptually a single pin could find its way in here with a pin collection wrapper around it.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/Origen-SDK/origen_testers/pull/11/files/cf95300ee23e359553fdbe52af837c573503a634#r60585418>